catroid/src/main/java/org/catrobat/catroid/sensing/CollisionInformation.java

```diff
@@ -78,11 +79,18 @@ public int getNumberOfVertices() {
 		return size;
 	}
 
+	public String getImageMimeType(String pathName) {
```

Comment:
since the `LookData` in pocket code is bascially a wrapper for image files and because it is always associated with one particular image file in storage it would make sense to move this method there. Then you can change it so that it does not need a parameter anymore such as:

```
public String getImageMimeType() {
    String pathName = file.getAbsolutePath();
    BitmapFactory.Options options = new BitmapFactory.Options();
    options.inJustDecodeBounds = true;
    BitmapFactory.decodeFile(pathName, options);
    return options.outMimeType;
}
```

then you can call it directly on the lookData in `loadOrCreateCollisionPolygon()`:

```
    ...
    if (!lookData.getImageMimeType().equals("image/png")) {
    ...
```